### ISSUE TITLE
--with-volumes option should be usable with stop and destroy

### ISSUE THREAD

**Issue by Lowess** [ansible/ansible-container#535]
##### ISSUE TYPE

 - Bug Report

##### container.yml 

```
version: "2"
settings:
  conductor_base: ubuntu:trusty
services:
  bug:
    from: ubuntu:trusty
    command: [ "/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait" ]
    roles:
      - debug
registries: {}

```

##### OS / ENVIRONMENT

```
Ansible Container, version 0.9.1rc0
Linux, latitude, 4.4.0-75-generic, #96~14.04.1-Ubuntu SMP Thu Apr 20 11:06:30 UTC 2017, x86_64
2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] /home/florian/.virtualenvs/gumref/bin/python
{
  "ContainersPaused": 0, 
  "Labels": null, 
  "CgroupDriver": "cgroupfs", 
  "ContainersRunning": 2, 
  "ContainerdCommit": {
    "Expected": "4ab9917febca54791c5f071a9d1f404867857fcc", 
    "ID": "4ab9917febca54791c5f071a9d1f404867857fcc"
  }, 
  "InitBinary": "docker-init", 
  "NGoroutines": 33, 
  "Swarm": {
    "Managers": 0, 
    "ControlAvailable": false, 
    "NodeID": "", 
    "Cluster": {
      "Spec": {
        "TaskDefaults": {}, 
        "Orchestration": {}, 
        "EncryptionConfig": {
          "AutoLockManagers": false
        }, 
        "Raft": {
          "HeartbeatTick": 0, 
          "ElectionTick": 0
        }, 
        "CAConfig": {}, 
        "Dispatcher": {}
      }, 
      "Version": {}, 
      "ID": "", 
      "CreatedAt": "0001-01-01T00:00:00Z", 
      "UpdatedAt": "0001-01-01T00:00:00Z"
    }, 
    "Nodes": 0, 
    "Error": "", 
    "RemoteManagers": null, 
    "LocalNodeState": "inactive", 
    "NodeAddr": ""
  }, 
  "LoggingDriver": "json-file", 
  "OSType": "linux", 
  "HttpProxy": "", 
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  }, 
  "DriverStatus": [
    [
      "Root Dir", 
      "/var/lib/docker/aufs"
    ], 
    [
      "Backing Filesystem", 
      "extfs"
    ], 
    [
      "Dirs", 
      "173"
    ], 
    [
      "Dirperm1 Supported", 
      "true"
    ]
  ], 
  "OperatingSystem": "Ubuntu 14.04.5 LTS", 
  "Containers": 8, 
  "HttpsProxy": "", 
  "BridgeNfIp6tables": true, 
  "MemTotal": 16728293376, 
  "SecurityOptions": [
    "name=apparmor"
  ], 
  "Driver": "aufs", 
  "IndexServerAddress": "https://index.docker.io/v1/", 
  "ClusterStore": "", 
  "InitCommit": {
    "Expected": "949e6fa", 
    "ID": "949e6fa"
  }, 
  "Isolation": "", 
  "SystemStatus": null, 
  "OomKillDisable": true, 
  "ClusterAdvertise": "", 
  "SystemTime": "2017-05-22T01:13:24.6808776-07:00", 
  "Name": "latitude", 
  "CPUSet": true, 
  "RegistryConfig": {
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ], 
    "IndexConfigs": {
      "docker.io": {
        "Official": true, 
        "Name": "docker.io", 
        "Secure": true, 
        "Mirrors": null
      }
    }, 
    "Mirrors": []
  }, 
  "DefaultRuntime": "runc", 
  "ContainersStopped": 6, 
  "NCPU": 4, 
  "NFd": 28, 
  "Architecture": "x86_64", 
  "KernelMemory": true, 
  "CpuCfsQuota": true, 
  "Debug": true, 
  "ID": "6QO6:QP4I:W7NZ:WYFR:L6FL:7J44:VKQP:BFWY:K6ME:T2ER:HEUT:4UCU", 
  "IPv4Forwarding": true, 
  "KernelVersion": "4.4.0-75-generic", 
  "BridgeNfIptables": true, 
  "NoProxy": "", 
  "LiveRestoreEnabled": false, 
  "ServerVersion": "17.03.1-ce", 
  "CpuCfsPeriod": true, 
  "ExperimentalBuild": false, 
  "MemoryLimit": true, 
  "SwapLimit": false, 
  "Plugins": {
    "Volume": [
      "local"
    ], 
    "Network": [
      "bridge", 
      "host", 
      "macvlan", 
      "null", 
      "overlay"
    ], 
    "Authorization": null
  }, 
  "Images": 179, 
  "DockerRootDir": "/var/lib/docker", 
  "NEventsListener": 0, 
  "CPUShares": true, 
  "RuncCommit": {
    "Expected": "54296cf40ad8143b62dbcaa1d90e520a2136ddfe", 
    "ID": "54296cf40ad8143b62dbcaa1d90e520a2136ddfe"
  }
}
{
  "KernelVersion": "4.4.0-75-generic", 
  "Arch": "amd64", 
  "BuildTime": "2017-03-27T17:10:36.401799458+00:00", 
  "ApiVersion": "1.27", 
  "Version": "17.03.1-ce", 
  "MinAPIVersion": "1.12", 
  "GitCommit": "c6d412e", 
  "Os": "linux", 
  "GoVersion": "go1.7.5"
}
```

##### SUMMARY

The `--with-volumes` option should be usable with `stop` and `destroy` commands as ansible.cfg is taken in account when those commands are invoked .

I am using Ansible with vault. The vault file is actually a shell script that relies on [credstash + AWS DynamoDB](https://github.com/fugue/credstash):

```
# ansible.cfg
 [defaults]
 vault_password_file = get-vault-key.sh
```

```
# get-vault-key.sh
 #!/usr/bin/env bash
 
 ENV_DATA=./env-data
  
 if [ -f ${ENV_DATA} ]
 then
   . "${ENV_DATA}"
 else
   exit 1
 fi
 
 credstash -t ${REGION}-CredStash-GumRef-${TEAM} get ansible-vault-key
```

We are grabbing the ansible vault secret from AWS DynamoDB so we need AWS cli credentials to be mounted in the conductor container. In production (Jenkins CI/CD) we do not have issues with AWS credentials because we are using IAM roles. But on a developer's laptop we lose the ability to  `stop` and `destroy` an ansible container project.

It's actually a wider issue. Basically anyone using Ansible vault with Ansible container can not invoke 
`stop` and `destroy` as they can not pass `--with-volumes`.

Here is the way I invoke the build on a project:
```
 ansible-container build --with-volumes \
     /home/florian/workspace/gumref-demo:/gumref-demo \ # Contain the source code of a nodejs project
     /home/florian/.aws:/root/.aws # Contains AWS cli credentials
```

##### STEPS TO REPRODUCE

Install the latest ansible-container version from develop and configure `ansible.cfg` with a vault password file:
```
# ansible.cfg
[defaults]
vault_password_file = get-vault-key.sh
```

Vault a variable file that is used in `container.yml` and invoke `ansible-container stop` or `ansible-container destroy`

##### EXPECTED RESULTS

Ansible container should accept the `--with-volumes` option for `stop` and `destroy` commands.

##### ACTUAL RESULTS

```
(gumref) 01:11:35-florian~/workspace/gumref/skel/bug (master)$ ansible-container stop --with-volumes ~/.aws:/root/.aws
usage: ansible-container [-h] [--debug] [--devel] [--engine ENGINE_NAME]
                         [--project-path BASE_PATH]
                         [--project-name PROJECT_NAME] [--var-file VAR_FILE]
                         [--no-selinux]
                         {run,help,deploy,stop,destroy,restart,init,version,build,install,push,import}
                         ...
ansible-container: error: unrecognized arguments: --with-volumes
(gumref) 01:11:39-florian~/workspace/gumref/skel/bug (master)$ ansible-container stop 
Parsing conductor CLI args.
Engine integration loaded. Preparing to stop all containers.	engine=Docker™ daemon
mount: warning: /home/florian/workspace/gumref/skel/bug/ansible-deployment/files seems to be mounted read-only.
mount: warning: /home/florian/workspace/gumref/skel/bug/ansible-deployment/templates seems to be mounted read-only.
Traceback (most recent call last):
  File "/usr/local/bin/credstash", line 11, in <module>
    load_entry_point('credstash==1.13.2', 'console_scripts', 'credstash')()
  File "/usr/local/bin/credstash.py", line 805, in main
    getSecretAction(args, region, **session_params)
  File "/usr/local/bin/credstash.py", line 244, in func_wrapper
    return func(*args, **kwargs)
  File "/usr/local/bin/credstash.py", line 407, in getSecretAction
    **session_params))
  File "/usr/local/bin/credstash.py", line 436, in getSecret
    KeyConditionExpression=boto3.dynamodb.conditions.Key("name").eq(name))
  File "/usr/local/lib/python2.7/dist-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 253, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 544, in _make_api_call
    operation_model, request_dict)
  File "/usr/local/lib/python2.7/dist-packages/botocore/endpoint.py", line 141, in make_request
    return self._send_request(request_dict, operation_model)
  File "/usr/local/lib/python2.7/dist-packages/botocore/endpoint.py", line 166, in _send_request
    request = self.create_request(request_dict, operation_model)
  File "/usr/local/lib/python2.7/dist-packages/botocore/endpoint.py", line 150, in create_request
    operation_name=operation_model.name)
  File "/usr/local/lib/python2.7/dist-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/botocore/signers.py", line 90, in handler
    return self.sign(operation_name, request)
  File "/usr/local/lib/python2.7/dist-packages/botocore/signers.py", line 154, in sign
    auth.add_auth(request)
  File "/usr/local/lib/python2.7/dist-packages/botocore/auth.py", line 339, in add_auth
    raise NoCredentialsError
botocore.exceptions.NoCredentialsError: Unable to locate credentials
ERROR! Vault password script /src/get-vault-key.sh returned non-zero (1): None
All services stopped.	playbook_rc=1
Conductor terminated. Cleaning up.	command_rc=0 conductor_id=b44d63bd45e93d68423148d795bf6e3608bff091c711c764422060c0bf2d6f66 save_container=False
```

**Comment by Lowess** [comment]
I am submitting a PR for this https://github.com/ansible/ansible-container/pull/536

**Comment by chouseknecht** [comment]
@Lowess 

Thanks for taking the time to open this issue and share your feedback.  

I added some comments to the PR.

**Comment by Lowess** [comment]
Hey @chouseknecht , let me cleanup my code change and the options you suggested.

**Comment by chouseknecht** [comment]
Fixed by #536
